packages/mobile/src/components/edit/PriceAndAudienceField/PriceAndAudienceScreen.tsx

```diff
@@ -186,13 +188,40 @@ export const PriceAndAudienceScreen = () => {
     )
   }, [dispatch])
 
+  // Listen for `navigation.goBack` events and in the case of going back
+  // reset the availability fields.
+  // Note that this is a stop gap against a better model which would be to have
+  // each radio subgroup manage its own state. That requires a larger refactor.
+  useEffect(() => {
+    const listener = navigation.addListener('beforeRemove', ({ data }) => {
```

Comment:
I'm curious if other sub-forms have this problem? or is it only this one that changes data before submission/save?

Reply:
yeah they all have this problem, but this is the one that hasn't covered all the edge cases. since its the most complex by far.